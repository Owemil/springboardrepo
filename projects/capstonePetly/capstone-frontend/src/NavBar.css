.oc-header {
    display: inline-block;
}

a {
    color: black;
    text-decoration: none;
}