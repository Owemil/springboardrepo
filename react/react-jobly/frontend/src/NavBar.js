import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import JoblyContext from "./JoblyContext";

/**NavBar component takes in user context and a logout funciton which empties both the state and localStorage for a user and their token
 * 
 * it renders a navbar to stay at the top of the screen thorughout the whole app
 */

function NavBar({ logout }) {
    const { user } = useContext(JoblyContext)

    return (
        <div>
            {user ?
                <nav>
                    <NavLink to='/'>
                        Jobly
                    </NavLink>
                    <NavLink to='/companies'>
                        companies
                    </NavLink>
                    <NavLink to='/jobs'>
                        Jobs
                    </NavLink>
                    <NavLink to='/profile'>
                        profile
                    </NavLink>
                    <NavLink onClick={logout}>
                        Log out {user.user.username}
                    </NavLink>
                </nav>
                :
                <nav>
                    <NavLink to='/'>
                        Jobly
                    </NavLink>
                    <NavLink to='/login'>
                        login
                    </NavLink>
                    <NavLink to='/signup'>
                        Sign up
                    </NavLink>
                </nav>}

        </div>
    )
}
export default NavBar