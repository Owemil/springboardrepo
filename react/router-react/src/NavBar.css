.navBar a {
    padding: 1em;
    color: black;
}

.navBar a.active {
    padding: 1em;
    color: goldenrod;
}