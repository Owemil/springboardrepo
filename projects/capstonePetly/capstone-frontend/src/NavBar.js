import React from 'react';
import useToggle from "./helpers/useToggle"
import {
  Navbar,
  NavbarBrand,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Button,
  Offcanvas,
  OffcanvasHeader,
  OffcanvasBody,
  CloseButton,
} from 'reactstrap';
import "./NavBar.css"
import { NavLink } from 'react-router-dom';

// NavBar component takes in a function to logout, user data and an array of other animal data
// it renders a navbar based on if a user is logged in or not

function NavBar({ logout, user, other }) {
  const [isOCOpen, toggleOC] = useToggle()
  return (
    <div>
      {user ?
        <Navbar >
          <NavbarBrand href="/">Petly</NavbarBrand>

          <NavLink to="/user">My Account</NavLink>

          <NavLink onClick={logout} to="/">Log out</NavLink>

          <div>
            <Button
              className="user-btn"
              color="primary"
              onClick={toggleOC}
            >
              Search
            </Button>
            <Offcanvas
              direction="end"
              isOpen={isOCOpen}
            >
              <OffcanvasHeader onClick={toggleOC}>

                <NavLink to='/' className='oc-header'>Petly</NavLink>

                <CloseButton className='oc-header'></CloseButton>
              </OffcanvasHeader>
              <OffcanvasBody>
                <div>

                  <h3>Welcome Back, {user.firstName}</h3>
                </div>
                <br />
                <h3>searchs</h3>

                <NavLink to="/search/Dogs">Dogs</NavLink>
                <br />
                <NavLink to="/search/Cats">Cats</NavLink>
                <br />
                <NavLink to="/search/Birds">Birds</NavLink>
                <br />
                <NavLink to="/search/Rabbits">Rabbits</NavLink>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Barnyard
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.barnyard.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Reptile
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.reptile.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Small & Furry
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.smallfurry.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Scales, Shells & Other
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.scalesshellsother.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <NavLink to="/orgs">Organizations</NavLink>
              </OffcanvasBody>
            </Offcanvas>
          </div>


        </Navbar>
        :
        <Navbar >
          <NavbarBrand href="/">Petly</NavbarBrand>

          <NavLink to="/signup">Signup</NavLink>

          <NavLink to="/login">Login</NavLink>

          <div>
            <Button
              className="user-btn"
              color="primary"
              onClick={toggleOC}
            >
              Search
            </Button>
            <Offcanvas
              scrollable
              direction="end"
              isOpen={isOCOpen}
            >
              <OffcanvasHeader onClick={toggleOC}>

                <NavLink to='/' className='oc-header'>Petly</NavLink>

                <CloseButton className='oc-header'></CloseButton>
              </OffcanvasHeader>
              <OffcanvasBody>
                <div>

                  <h3>Welcome, Guest</h3>
                </div>
                <br />
                <h3>searchs</h3>

                <NavLink to="/search/Dogs">Dogs</NavLink>
                <br />
                <NavLink to="/search/Cats">Cats</NavLink>
                <br />
                <NavLink to="/search/Birds">Birds</NavLink>
                <br />
                <NavLink to="/search/Rabbits">Rabbits</NavLink>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Barnyard
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.barnyard.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Reptile
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.reptile.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Small & Furry
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.smallfurry.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <UncontrolledDropdown>
                  <DropdownToggle nav caret>
                    Scales, Shells & Other
                  </DropdownToggle>
                  <DropdownMenu end>
                    {other.scalesshellsother.map((item, idx) => {
                      return (
                        <>
                          <NavLink to={`/search/${item}`} key={idx}>{item}</NavLink>
                          <br />
                        </>
                      )
                    })}
                  </DropdownMenu>
                </UncontrolledDropdown>

                <NavLink to="/orgs">Organizations</NavLink>

              </OffcanvasBody>
            </Offcanvas>
          </div>
        </Navbar>}

    </div>
  );
}

export default NavBar;