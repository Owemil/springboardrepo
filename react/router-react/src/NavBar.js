import React from "react";
import { NavLink } from "react-router-dom";
import "./NavBar.css"

function NavBar() {

    return (
        <nav className="navBar">
            <NavLink to="/cheezit">Cheezit</NavLink>
            <NavLink to="/cosmic">Cosmic Brownie</NavLink>
            <NavLink to="/milk">Milk</NavLink>
        </nav>
    )
}

export default NavBar