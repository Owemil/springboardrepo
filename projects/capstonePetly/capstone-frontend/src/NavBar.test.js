import React, { act } from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar.js";


const user = { firstName: "Johnny" }
const nullUser = null
const otherFilters = {
    filterSet: new Set(["Barnyard", "Reptile", "Small & Furry", "Scales, Shells & Other"]),
    scalesshellsother: ["Fish", "Frogs", "Turtles", "Tortoises",],
    reptile: ["Geckos", "Iguanas", "Snakes", "Lizards"],
    smallfurry: ["Chinchillas", "Ferrets", "Gerbils", "Hamsters", "Mice", "Rats",],
    barnyard: ["Chickens", "Cows", "Donkeys", "Ducks", "Geese", "Goats", "Pigs", "Ponies", "Turkeys",]
}
const logout = jest.fn()
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));


it("should render user", function () {
    render(
        <MemoryRouter>
            <NavBar logout={logout} user={user} other={otherFilters} />
        </MemoryRouter>)
})
it("should render nullUser", function () {
    render(
        <MemoryRouter>
            <NavBar logout={logout} user={nullUser} other={otherFilters} />
        </MemoryRouter>)
})

it("should match snapshot user", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <NavBar logout={logout} user={user} other={otherFilters} />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot nullUser", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <NavBar logout={logout} user={nullUser} other={otherFilters} />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})

it("should show login/signup when no user", function () {

    const { getByText } = render(
        <MemoryRouter>
            <NavBar logout={logout} user={nullUser} other={otherFilters} />
        </MemoryRouter>)
    expect(getByText('Login')).toBeInTheDocument()
    expect(getByText('Signup')).toBeInTheDocument()
})
it("should show logout/my account when user", function () {

    const { getByText } = render(
        <MemoryRouter>
            <NavBar logout={logout} user={user} other={otherFilters} />
        </MemoryRouter>)
    expect(getByText('Log out')).toBeInTheDocument()
    expect(getByText('My Account')).toBeInTheDocument()
})
it("should show guest msg when no user", function () {

    const { getByText } = render(
        <MemoryRouter>
            <NavBar logout={logout} user={nullUser} other={otherFilters} />
        </MemoryRouter>)

    fireEvent.click(getByText('Search'))

    expect(getByText('Welcome, Guest')).toBeInTheDocument()
})
it("should show user msg when user", function () {

    const { getByText } = render(
        <MemoryRouter>
            <NavBar logout={logout} user={user} other={otherFilters} />
        </MemoryRouter>)

    fireEvent.click(getByText('Search'))

    expect(getByText('Welcome Back, Johnny')).toBeInTheDocument()
})  
