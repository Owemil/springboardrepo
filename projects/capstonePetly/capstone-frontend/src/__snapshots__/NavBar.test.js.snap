// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot nullUser 1`] = `[Function]`;

exports[`should match snapshot user 1`] = `[Function]`;
